vmm/src/serial_buffer.rs

```diff
@@ -28,14 +30,22 @@ impl SerialBuffer {
             write_out,
         }
     }
+
+    fn fill_buffer(&mut self, buf: &[u8]) {
+        if self.buffer.len() + buf.len() > MAX_BUFFER_SIZE {
+            let num_bytes_to_remove = self.buffer.len() + buf.len() - MAX_BUFFER_SIZE;
+            self.buffer.drain(..num_bytes_to_remove);
```

Comment:
I think there is a chance that `num_bytes_to_remove` is larger than `self.buffer.len()`, which will be a out-of-bound access?

Reply:
Yes you're right, the `buf` could be already bigger than `MAX_BUFFER_SIZE`.